src/js/aos.js

```diff
@@ -48,16 +49,21 @@ const isBrowserNotSupported = () => document.all && !window.atob;
 const initializeScroll = function initializeScroll() {
   // Extend elements objects in $aosElements with their positions
   $aosElements = prepare($aosElements, options);
+  // Define container element
+  let container =
+    options.container == 'window'
+      ? window
+      : document.querySelector(options.container) || window;
```

Comment:
I don't think it's a great idea to only take `String` to target an element. Since this is executed synchronously, we should at least support a `HTMLElement`, if not only it.

Please use `window` (instead of `'window'`) for the default of `container`, and only search for the element if `options.container` is a string.

Reply:
Also: if we have no results with `document.querySelector(options.container)`, there is something wrong because we cannot find the element the user want to target with the provided `container` option. We should throw an error in this case.

Reply:
Take a look and tell me what to add/change/fix/rename

```js
// Define container element
let container = isElementValid(options.container);
if (!container) throw `[AOS] Defined property 'container' is not valid`;

(...)

const isElementValid = function isElementValid(container) {
  if (container instanceof Element || container == window) return container;
  if (typeof container === 'string') {
    const queryResult = document.querySelector(container);
    if (queryResult) return queryResult;
  }
  return false;
};
```

And thank you sooo much for your time and help :)

Reply:
and, of course, `let` -> `const` ;)

Reply:
The error message must give the user entry, tell what is wrong and what is expected.
```diff
- [AOS] Defined property 'container' is not valid
+ AOS - cannot find the container element "xxx". The container option must be an HTMLElement or a CSS Selector.
```

This function doesn't returns a Boolean but performs a search. Use `find...` instead of `is...`.
```diff
- let container = isElementValid(options.container);
+ let container = findElement(options.container);
```

This function returns the element if it is found. Returns nothing if nothing is found.
```diff
- return false;
+ return null;
```

Also, add some documentation on top of this helper :)